### PR TITLE
QDOC: Provide links to external documentation

### LINKED ISSUE #1875: Quick documentaion should contain link to the on-line docs
PyCharm has a useful feature where it adds an "View external documentaion" link to the quick-doc popup (<kbd>Ctrl+Q</kbd>):

image

I think we should do the same for Rust, using `docs.rs` and `https://doc.rust-lang.org/std/`. 

The entry point for the relevant feature is `org.rust.ide.docs.RsDocumentationProvider`.

### PR DESCRIPTION
**Pull request by Undin**
It's initial implementation - it provides links only for stdlib items and it doesn't support reexports (as well as all our quick documentation does).
It means:
* we generate link according to a fully qualified name of resolved item and don't take into account its reexport path. For example, we generate https://doc.rust-lang.org/alloc/string/struct.String.html instead of https://doc.rust-lang.org/std/string/struct.String.html. It's not a big problem, because doc.rust-lang.org provides very similar pages for both links.
* we don't provide links to element from private module but available only via reexport. For example, `Iterator` trait.

Also, it's the first part of big plan to generate fully qualified name with similar schema as documentation link.

Part of #1875

### CODE REVIEW

**Comment by mehcode** [comment]
Why is the focus on external documentation when easily 90% of rust users have documentation downloaded locally?

File: src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt [hunk 1]
```diff
@@ -14,3 +19,162 @@ val RsQualifiedNamedElement.qualifiedName: String? get() {
...
+                    when (owner) {
+                        is RsAbstractableOwner.Trait -> owner.trait
+                        is RsAbstractableOwner.Impl -> {
+                            return owner.impl.typeReference?.toParentItem()
+                        }
+                        else -> return null
+                    }
+                }
+                is RsEnumVariant -> element.parentEnum
+                is RsFieldDecl -> element.parentStruct
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt]: Should we take into account that enums can contain fields too?

**Comment by Undin** [comment]
@mehcode 
Yeah, it makes sense and we will implement such behavior eventually
But local link creation requires some infrastructural changes and we will make them in separate PR

File: src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt (hunk 1 above)
Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt]: It's an interesting question
On the one hand, named field in enum variant can have qualified name
On the other hand, rustdoc doesn't add info about it in a documentation
So, let's do same things as rustdoc and don't generate qualified name for fields in enum variant now and will implement it later if it is necessary

**Comment by Undin** [comment]
bors r=vlad20012

**Pull request merged**